internal/controller/promotionstrategy_controller.go

```diff
@@ -296,11 +293,6 @@ func (r *PromotionStrategyReconciler) createOrUpdatePreviousEnvironmentCommitSta
 	gvk := promoterv1alpha1.GroupVersion.WithKind(kind)
 	controllerRef := metav1.NewControllerRef(ctp, gvk)
 
-	branch := "no previous environment"
-	if previousEnvironmentStatus != nil {
```

Comment:
The nil case could never be hit. We don't call this function unless there's a previous env.